compiler/ballerina-parser/src/test/java/io/ballerinalang/compiler/parser/test/syntax/statements/WhileStatementTest.java

```diff
@@ -36,6 +36,11 @@ public void testWhileWithBody() {
         testFile("while-stmt/while_stmt_source_02.bal", "while-stmt/while_stmt_assert_02.json");
     }
 
+    @Test
+    public void testWhileWithOnFailClause() {
```

Comment:
Shall we add some negative tests as well?
eg: missing certain parts of the on-fail-clause, having additional tokens in between the tokens in on-fail-clause, etc.

Reply:
We'll improve error handling and negative test cases separately. Created https://github.com/ballerina-platform/ballerina-lang/issues/25711 to track the progress.